applications/swimming_DEM_application/python_scripts/swimming_DEM_procedures.py

```diff
@@ -456,7 +457,7 @@ def __init__(self,
                  rigid_faces_model_part,
                  mixed_model_part):
 
-        self.gid_io                 = gid_io
+        self.gid_io                 = weakref.proxy(gid_io)
```

Comment:
That leak was inside gid_io class (c++ code), it was already fixed in #1074

Reply:
Hello, I know about it. I ran the DEM case I had and it worked without leaking, however, the leak remained in my swimming_DEM_app cases (I double checked this to be the case when running the case with the master version). After adding a few weakrefs (which I think make sense anyways) the leak has vanished.

Reply:
Oh, ok, I thought that would have solve all problems with the gid_io, seems i was wrong sorry